container/docker/factory.go

```diff
@@ -273,6 +269,9 @@ func Register(factory info.MachineInfoFactory, fsInfo fs.FsInfo) error {
 				return fmt.Errorf("cAdvisor requires docker version %v or above but we have found version %v reported as \"%v\"", expected_version, dockerVersion, version_string)
 			}
 		}
+		if dockerVersion[0] == 1 && dockerVersion[1] >= 10 {
```

Comment:
This is not always true since we can enable systemd support optionally. How do we handle that?